### ISSUE TITLE
I compared Ryzen 7 2700X + GTX1080ti and Xeon 5690 + GTX1080ti.

### ISSUE THREAD

**Issue by sounansu** [AlexeyAB/darknet#1775]
I bought Ryzen 7 2700X for renewal of my PC.
Old were 2 Sockets of  Xeon 5690 (Server MB). It was nice but sometimes dumped core......

So, I compared new and old machine by Yolov3-tiny training.( VOC data)
Old machine
Oct 16 01:09 darknet_alexeyab_Xeon5690/backup/yolov3-tiny-voc_50000.weights
Oct 16 00:05 darknet_alexeyab_Xeon5690/backup/yolov3-tiny-voc_40000.weights
Oct 15 22:59 darknet_alexeyab_Xeon5690/backup/yolov3-tiny-voc_30000.weights
Oct 15 21:54 darknet_alexeyab_Xeon5690/backup/yolov3-tiny-voc_20000.weights
Oct 15 20:48 darknet_alexeyab_Xeon5690/backup/yolov3-tiny-voc_10000.weights

New machine
Oct 17 05:13 darknet_alexeyab_Ryzen2700X/backup/yolov3-tiny-voc_50000.weights
Oct 17 03:30 darknet_alexeyab_Ryzen2700X/backup/yolov3-tiny-voc_40000.weights
Oct 17 01:47 darknet_alexeyab_Ryzen2700X/backup/yolov3-tiny-voc_30000.weights
Oct 17 00:03 darknet_alexeyab_Ryzen2700X/backup/yolov3-tiny-voc_20000.weights
Oct 16 22:20 darknet_alexeyab_Ryzen2700X/backup/yolov3-tiny-voc_10000.weights

Oh!, New machine is slow.....

Is Zen arch. not suitable for Deep learning?

**Comment by kmsravindra** [comment]
Could you show us the comparison metric and the fps for each of these GPUs?

**Comment by AlexeyAB** [comment]
@sounansu 

* What parameters did you use in the Makefile? Did you use `OPENCV=1`?

* What CPU usage was during training on a new machine?

* What CUDA and cuDNN versions did you use in both cases? Show result of commands `nvcc --version` and `nvidia-smi`

**Comment by sounansu** [comment]
@AlexeyAB 
Sorry, my xeon machine was not here! It was in the Bin!

- I only change `AVX=` parameter for Xeon. It lack of AVX.

- I always use `OPENCV=1`

- CUDA and cuDNN versions are CUDA 10 and cuDNN 7.3.1

- nvidia-smi and CPU usage were below



So,I try to change thread parameter of your `detector.c`.

```
$ git diff *
diff --git a/src/detector.c b/src/detector.c
index 5926979..39bda25 100644
--- a/src/detector.c
+++ b/src/detector.c
@@ -108,7 +108,7 @@ void train_detector(char *datacfg, char *cfgfile, char *weightfile, int *gpus, i
     args.small_object = net.small_object;
     args.d = &buffer;
     args.type = DETECTION_DATA;
-    args.threads = 16;    // 64
+    args.threads = 64;    // 64

     args.angle = net.angle;
     args.exposure = net.exposure;
@@ -116,7 +116,7 @@ void train_detector(char *datacfg, char *cfgfile, char *weightfile, int *gpus, i
     args.hue = net.hue;

 #ifdef OPENCV
-    args.threads = 3 * ngpus;
+    args.threads = 12 * ngpus;
     IplImage* img = NULL;
     float max_img_loss = 5;
     int number_of_lines = 100;
@@ -389,7 +389,7 @@ void validate_detector(char *datacfg, char *cfgfile, char *weightfile, char *out
     float thresh = .005;
     float nms = .45;

-    int nthreads = 4;
+    int nthreads = 16;
     image *val = calloc(nthreads, sizeof(image));
     image *val_resized = calloc(nthreads, sizeof(image));
     image *buf = calloc(nthreads, sizeof(image));
@@ -598,7 +598,7 @@ void validate_detector_map(char *datacfg, char *cfgfile, char *weightfile, float
     const float nms = .45;
     const float iou_thresh = 0.5;

-    int nthreads = 4;
+    int nthreads = 16;
     image *val = calloc(nthreads, sizeof(image));
     image *val_resized = calloc(nthreads, sizeof(image));
     image *buf = calloc(nthreads, sizeof(image));
```
and training again.

Oct 17 17:07 darknet_alexeyab_Ryzen2700X_more_thread/backup/yolov3-tiny-voc_50000.weights
Oct 17 16:13 darknet_alexeyab_Ryzen2700X_more_thread/backup/yolov3-tiny-voc_40000.weights
Oct 17 15:18 darknet_alexeyab_Ryzen2700X_more_thread/backup/yolov3-tiny-voc_30000.weights
Oct 17 14:23 darknet_alexeyab_Ryzen2700X_more_thread/backup/yolov3-tiny-voc_20000.weights
Oct 17 13:30 darknet_alexeyab_Ryzen2700X_more_thread/backup/yolov3-tiny-voc_10000.weights

It is very fast! It is faster than Xeon! Thank you @AlexeyAB ! Your darknet fork is very nice!
---------------
F.Y.I.

I only can show results of nvidia-smi command and top command of 2700X, but original and more thread . 

nvidia-smi of more thread
```
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 410.48                 Driver Version: 410.48                    |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce GTX 108...  Off  | 00000000:1D:00.0 Off |                  N/A |
| 38%   67C    P2   218W / 250W |   4133MiB / 11176MiB |     91%      Default |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|    0     32658      C   ./darknet                                   4123MiB |
+-----------------------------------------------------------------------------+

```
top command of more thread
```
top - 18:43:41 up 22:34,  2 users,  load average: 10.53, 7.12, 3.19
Tasks: 243 total,   3 running, 143 sleeping,   0 stopped,   0 zombie
%Cpu(s): 23.7 us,  1.0 sy,  0.0 ni, 75.3 id,  0.0 wa,  0.0 hi,  0.0 si,  0.0 st
KiB Mem : 65981752 total, 28840024 free,  3453588 used, 33688140 buff/cache
KiB Swap: 67041272 total, 67041272 free,        0 used. 61792972 avail Mem

  PID USER      PR  NI    VIRT    RES    SHR S  %CPU %MEM     TIME+ COMMAND
32658 sounansu  20   0 26.273g 3.215g 402948 R  1369  5.1  54:19.42 darknet

```
nvidia-smi of original code
```
Wed Oct 17 18:47:48 2018
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 410.48                 Driver Version: 410.48                    |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce GTX 108...  Off  | 00000000:1D:00.0 Off |                  N/A |
| 30%   63C    P2    78W / 250W |   2099MiB / 11176MiB |     23%      Default |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|    0     19689      C   ./darknet                                   2089MiB |
+-----------------------------------------------------------------------------+

```
top of original code
```
top - 18:48:49 up 22:40,  2 users,  load average: 4.01, 5.30, 3.58
Tasks: 247 total,   1 running, 144 sleeping,   0 stopped,   0 zombie
%Cpu(s): 23.7 us,  1.0 sy,  0.0 ni, 75.3 id,  0.0 wa,  0.0 hi,  0.0 si,  0.0 st
KiB Mem : 65981752 total, 30359684 free,  1842876 used, 33779192 buff/cache
KiB Swap: 67041272 total, 67041272 free,        0 used. 63403672 avail Mem

  PID USER      PR  NI    VIRT    RES    SHR S  %CPU %MEM     TIME+ COMMAND
19689 sounansu  20   0 20.676g 1.513g 224276 S 387.5  2.4   5:06.82 darknet

```

**Comment by AlexeyAB** [comment]
@sounansu 
So with the same GPU GTX1080ti:

* 2 Sockets of Xeon 5690 Server - took **65 minutes** for each 10 000 iterations
* AMD Ryzen 7 2700X - took **55 minutes** for each 10 000 iterations - if this line is un-commented: https://github.com/AlexeyAB/darknet/blob/3b59178dec5e9b105b2fba23001c768206da0276/src/detector.c#L120

Can you get this good speed if you use `args.threads = 6 * ngpus;` instead of `args.threads = 12 * ngpus;`? Or 12 is the best value in your case?

**Comment by dreambit** [comment]
@AlexeyAB Hi, i have intel i5 2300, gtx 1080ti, 16gb ram; opencv, cudnn are enabled.
It takes ~1 hour for 1000 iteration, batch size = 64, sub = 32.

Compared to

> 2 Sockets of Xeon 5690 Server - took 65 minutes for each 10 000 iterations
AMD Ryzen 7 2700X - took 55 minutes for each 10 000 iterations - if this line is un-commented:

my results are not so good ;(

**Comment by AlexeyAB** [comment]
> @AlexeyAB Hi, i have intel i5 2300, gtx 1080ti, 16gb ram; opencv, cudnn are enabled. Network size is 608x608
> It takes **~1 hour for 1000 iteration**, batch size = 64, sub = 32.
> 
> Compared to
> 
> 2 Sockets of Xeon 5690 Server - took **65 minutes for each 10 000 iterations**
> AMD Ryzen 7 2700X - took **55 minutes for each 10 000 iterations** - if this line is un-commented:

He uses Yolov3-tiny, while you use Yolov3 full model.

Try to set `batch= 64 subdivisions=16` the training should go faster without Out of memory errors. 
`yolov3.cfg` with `batch= 64 subdivisions=16 width=416 height=416 random=1` should take 10.6 GB of GPU RAM, while your GTX 1080 Ti has 11 GB GPU RAM.

**Comment by dreambit** [comment]
@AlexeyAB Thanks, yes it is much faster, ~8 GB is used, but i hope that accuracy will not suffer.

**Comment by AlexeyAB** [comment]
@dreambit In general the higher mini_batch = batch/subdivisions, the higher speed and slighltly higher accuracy
